src/routes/+layout.svelte

```diff
@@ -29,20 +40,66 @@
 
   let navMenuExpanded = false;
   $: if ($navigating) navMenuExpanded = false;
+
+  // This is a workaround for https://github.com/sveltejs/kit/issues/10122
+  let lastPageURL: URL | undefined, lastPageError: App.Error | null | undefined;
+  $: ({ url: lastPageURL, error: lastPageError } = $page || {});
+  beforeNavigate(async ({ to, type, cancel }) => {
+    const previousPreviewValue = lastPageURL?.searchParams.get("preview");
+    if (
+      typeof previousPreviewValue === "string" &&
+      type === "link" &&
+      to &&
+      !to.url.searchParams.has("preview") &&
+      !lastPageError &&
+      !previewAuthenticationError &&
+      !(to.url.pathname === "/login" || to.url.pathname === "/logout")
+    ) {
+      to.url.searchParams.set("preview", previousPreviewValue);
+    }
+
+    if (
+      type === "link" &&
+      to &&
+      previewAuthenticationError &&
+      !(to.url.pathname === "/login" || to.url.pathname === "/logout")
+    ) {
+      to.url.searchParams.delete("preview");
+      cancel();
+      await goto(to.url.toString());
+      invalidate("app:previewAuthenticationError");
+    }
+  });
+
+  $: inIframe = browser && isInIframe();
+  $: loginLinkProps = inIframe ? { target: "_blank" } : {};
 </script>
 
 <RootIntersectionObserver enabled={intersectionObserverSupport && !lazyImageLoadingSupport}>
   <a class="usa-skipnav" href="#main-content">Skip to main content</a>
   <Banner />
   <div class="usa-overlay" />
   <Header {navItems} {secondaryNavItems} {siteTitle} bind:navMenuExpanded />
-  <slot />
-  <!-- TODO: reinstate <Footer /> once designed and implemented -->
-  <footer class="usa-footer padding-top-7 padding-bottom-1 text-center">
-    ©2023 Louisiana Department of Agriculture and Forestry. All rights reserved
-  </footer>
-  <!-- TODO: reinstate <Identifier /> once designed and implemented -->
-  <!-- (or remove if encapsulated by footer) -->
+  {#if previewAuthenticationError}
+    <div class="usa-section usa-prose grid-container">
+      <h1>{previewAuthenticationError.status}</h1>
+      <p>{previewAuthenticationError.message}</p>
+      <p>
+        (Do you need to <LoginLink {...loginLinkProps}>log in</LoginLink> or visit the
+        <a href={$page.url.toString()}>public version of the page</a>?)
+      </p>
+    </div>
+  {:else}
+    <main id="main-content">
+      <slot />
+    </main>
+    <!-- TODO: reinstate <Footer /> once designed and implemented -->
+    <footer class="usa-footer padding-top-7 padding-bottom-1 text-center">
+      ©2023 Louisiana Department of Agriculture and Forestry. All rights reserved
+    </footer>
+    <!-- TODO: reinstate <Identifier /> once designed and implemented -->
+    <!-- (or remove if encapsulated by footer) -->
+  {/if}
```

Comment:
nit: we should probably still include the footer when there's a `previewAuthenticationError` right?